cachito/web/models.py

```diff
@@ -101,6 +102,36 @@ def get_final_states():
         """
         return ["complete", "failed"]
 
+    @staticmethod
+    def allow_transition(from_state: Optional[int], to_state: int) -> bool:
+        """Check if the state transition is allowed.
+
+        :param from_state: transition from this state.
+        :type from_state: int or None
+        :param int to_state: transition to this state.
+        :return: True if the transition is allowed, otherwise False is returned.
+        :rtype: bool
+        """
+        if from_state is None:
+            return True  # Allow to set the initial state.
```

Comment:
IMO `from_state` should not be `None`. Instead, this method should only be called if there already is a previous state.